MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/annotations/ClickOnLayerActivity.java

```diff
@@ -0,0 +1,153 @@
+package com.mapbox.mapboxandroiddemo.examples.annotations;
+
+import android.graphics.PointF;
+import android.graphics.RectF;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.style.layers.FillLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+import com.mapbox.services.commons.geojson.Feature;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillOpacity;
+
+/**
+ * Detect click events on a polygon that was added as a GeoJsonSource.
+ */
+
+public class ClickOnLayerActivity extends AppCompatActivity {
+
+  private MapView mapView;
+  private String geoJsonSourceId = "GeoJSON source";
+  private String geoJsonLayerId = "GeoJSON layer";
+  private FillLayer layer;
+  private GeoJsonSource source;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_click_on_layer);
+
+    Toast.makeText(ClickOnLayerActivity.this, R.string.click_on_polygon_toast_instruction,
+        Toast.LENGTH_SHORT).show();
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(final MapboxMap mapboxMap) {
+
+        try {
+
+          // Load GeoJSONSource
+          source = new GeoJsonSource(geoJsonSourceId, new URL("https://gist.githubusercontent"
+              + ".com/tobrun/cf0d689c8187d42ebe62757f6d0cf137/raw/4d8ac3c8333f1517df9d303"
+              + "d58f20f4a1d8841e8/regions.geojson"));
+
+          // Add GeoJsonSource to map
+          mapboxMap.addSource(source);
+
+        } catch (MalformedURLException malformedUrlException) {
+          Log.d("Invalid URL", "malformedUrlException = " + malformedUrlException.toString());
```

Comment:
don't see a technical reason to prefer `=` over `:`.

When logging exceptions you should use the 3 args Log methods:
```
Log.e(String tag, String msg, Throwable tr) 
```
No need for concatenation, overall a bit cleaner imo